### ISSUE TITLE
Add support for m.tombstone events in the matrix connector

### ISSUE THREAD

**Issue by Cadair** [opsdroid/opsdroid#1264]
When a matrix room is upgraded a `m.tombstone` event is emitted, with a link to the upgraded room. The matrix connector should automatically follow this link and join the new room and continue as normal.

**Comment by jacobtomlinson** [comment]
What happens if a bot misses this event (restart, downtime, etc)? Is there a way it can also query a room on startup to see if it has been upgraded?

**Comment by Cadair** [comment]
Yeah, I meant to add that, it should in theory come down in the initial sync everytime the bot starts, so it should also check for that as well.

**Comment by jacobtomlinson** [comment]
Ok! Sounds good.

**Comment by himanshupathak21061998** [comment]
Hey @Cadair I want to work on this issue can you please explain that I need to add extra event connector in this https://github.com/opsdroid/opsdroid/blob/7282a5da065cbe73212245f55b7cc4b3537ebd37/opsdroid/connector/matrix/events.py#L3  or anything else we have to do.

**Comment by Cadair** [comment]
There are a few components to this.

1) Emitting and sending tombstone events. This requires you to add an event class in `matrix/events.py` modify the event creator to emit that event as it comes down sync, and then write a handler on the connector to send a tombstone state event into the room.
2) Reacting to room upgrades. The real reason for tombstone events is almost always that the room has been upgraded. (See https://matrix.org/docs/spec/client_server/latest#id160) What we want to happen when the room is upgraded is that the bot seamlessly joins the new room and carries on.
3) Emitting room upgrades. This is probably more niche. The bot would need to do be able to send the POST api call for room upgrades. Sending a tombstone event isn't the same as upgrading the room, so we would probably want a new event for an upgrade. Also receiving a tombstone, isn't *always* an upgrade, so we would need to think about that as well.

**Comment by Cadair** [comment]
Implementing this also requires adding support for it in the matrix-nio library.
